shell/common/shell.cc

```diff
@@ -508,11 +508,13 @@ std::unique_ptr<Shell> Shell::Spawn(
         is_gpu_disabled));
     return result;
   };
-  std::unique_ptr<Shell> result;
+  // It's safe to store this value since it is set on the platform thread.
+  bool isGpuDisabled = false;
   GetIsGpuDisabledSyncSwitch()->Execute(
       fml::SyncSwitch::Handlers()
-          .SetIfFalse([&] { result = shell_maker(false); })
-          .SetIfTrue([&] { result = shell_maker(true); }));
+          .SetIfFalse([&isGpuDisabled] { isGpuDisabled = false; })
+          .SetIfTrue([&isGpuDisabled] { isGpuDisabled = true; }));
+  std::unique_ptr<Shell> result = shell_maker(isGpuDisabled);
```

Comment:
The shell_maker lambda can be removed and the contents can be moved here

Reply:
Done